File: app/src/main/java/com/ireddragonicy/konabessnext/ui/adapters/ActionCardAdapter.java
package com.ireddragonicy.konabessnext.ui.adapters;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.google.android.material.card.MaterialCardView;

import java.util.List;

import com.ireddragonicy.konabessnext.R;

public class ActionCardAdapter extends RecyclerView.Adapter<ActionCardAdapter.ViewHolder> {

    public static class ActionItem {
        public int iconResId;
        public String title;
        public String description;
        public boolean enabled;

        public ActionItem(int iconResId, String title, String description) {
            this(iconResId, title, description, true);
        }

        public ActionItem(int iconResId, String title, String description, boolean enabled) {
            this.iconResId = iconResId;
            this.title = title;
            this.description = description;
            this.enabled = enabled;
        }
    }

    private final List<ActionItem> items;
    private OnItemClickListener clickListener;

    public interface OnItemClickListener {
        void onItemClick(int position);
    }

    public ActionCardAdapter(List<ActionItem> items) {
        this.items = items;
    }

    public void setOnItemClickListener(OnItemClickListener listener) {
        this.clickListener = listener;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.action_item_card, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        ActionItem item = items.get(position);
        holder.icon.setImageResource(item.iconResId);
        holder.title.setText(item.title);
        holder.description.setText(item.description);
        holder.card.setEnabled(item.enabled);
        holder.card.setClickable(item.enabled);
        holder.card.setFocusable(item.enabled);
        
        // Simple disabled visual state
        holder.card.setAlpha(item.enabled ? 1f : 0.5f);

        holder.card.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if (clickListener == null) {
                    return;
                }
                int adapterPosition = holder.getBindingAdapterPosition();
                if (adapterPosition == RecyclerView.NO_POSITION) {
                    return;
                }
                ActionItem actionItem = items.get(adapterPosition);
                if (actionItem.enabled) {
                    clickListener.onItemClick(adapterPosition);
                }
            }
        });
    }

    @Override
    public int getItemCount() {
        return items.size();
    }

    static class ViewHolder extends RecyclerView.ViewHolder {
        MaterialCardView card;
        ImageView icon;
        TextView title;
        TextView description;

        ViewHolder(View itemView) {
            super(itemView);
            card = (MaterialCardView) itemView;
            icon = itemView.findViewById(R.id.icon);
            title = itemView.findViewById(R.id.title);
            description = itemView.findViewById(R.id.description);
        }
    }
}

File: app/src/main/java/com/ireddragonicy/konabessnext/ui/adapters/ActionCardAdapter.kt
package com.ireddragonicy.konabessnext.ui.adapters

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.google.android.material.card.MaterialCardView
import com.ireddragonicy.konabessnext.R

class ActionCardAdapter(private val items: List<ActionItem>) : RecyclerView.Adapter<ActionCardAdapter.ViewHolder>() {

    class ActionItem(
        val iconResId: Int,
        val title: String,
        val description: String,
        @JvmField val enabled: Boolean = true
    )

    fun interface OnItemClickListener {
        fun onItemClick(position: Int)
    }

    private var clickListener: OnItemClickListener? = null

    fun setOnItemClickListener(listener: OnItemClickListener) {
        this.clickListener = listener
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.action_item_card, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val item = items[position]
        holder.icon.setImageResource(item.iconResId)
        holder.title.text = item.title
        holder.description.text = item.description
        holder.card.isEnabled = item.enabled
        holder.card.isClickable = item.enabled
        holder.card.isFocusable = item.enabled

        // Simple disabled visual state
        holder.card.alpha = if (item.enabled) 1f else 0.5f

        holder.card.setOnClickListener {
            if (clickListener != null) {
                val adapterPosition = holder.bindingAdapterPosition
                if (adapterPosition != RecyclerView.NO_POSITION) {
                    val actionItem = items[adapterPosition]
                    if (actionItem.enabled) {
                        clickListener!!.onItemClick(adapterPosition)
                    }
                }
            }
        }
    }

    override fun getItemCount(): Int {
        return items.size
    }

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val card: MaterialCardView = itemView as MaterialCardView
        val icon: ImageView = itemView.findViewById(R.id.icon)
        val title: TextView = itemView.findViewById(R.id.title)
        val description: TextView = itemView.findViewById(R.id.description)
    }
}
